### DISCUSSION TITLE
Dafang Hacks e MQTT

### DISCUSSION THREAD

**Discussion by gforment** [indomus/forum#201] [HUB: Home Assistant]
Ciao, avevo in garage una vecchia telecamera  DAFANG e sono riuscito a caricare il software Dafang Hacks (https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks). Funziona tutto, autodiscovery MQTT compreso che mi crea una decina di entità fra switch e altro. 
Mi chiedevo se qualcuno invece era riuscito a integrare tramite MQTT manualmente (preferisco questa modalità che consente maggior controllo) e se poteva condividere la configurazione o in alternativa se c'è qualcuno che sa come recuperare una configurazione dalle entità create con autodiscovery.
Grazie!

**Comment by snakuzzo** [comment]
Non ho questa telecamera, ma credo tu possa farlo mettendoti in ascolto sulla coda del broker MQTT.
Guarda [QUI](https://indomus.it/formazione/mosquitto-mqtt-broker-comandi-utili/).

Se non vuoi andare ad indagare nel codice sorgente del progetto, puoi metterti in ascolto e vedi cosa passa quando dai un comando, cosa manda la telecamera...ed in base ai topic e payload ti crei le tue entità

**Reply by gforment** [reply]
Più semplice del previsto: l'autodiscovery della telecamera pubblica su un topic specifico le esatte configurazioni MQTT ma in formato JSON. Mi è bastato andare di copia/incolla e conversione in configurazione YAML. Consiglio anche l'uso di MQTT Explorer per la lettura delle code in modo più fruibile della riga di comando :)
Grazie!

**Reply by snakuzzo** [reply]
Che t'avevo detto ? ;)
Cmq meglio da riga di comando a mio avviso

**Reply by snakuzzo** [reply]
MQTT Explorer è già conosciutissimo

**Reply by gforment** [reply]
Il tema dell'autodiscovery risolve il problema in modo diverso senza dover fare reverse-engineering dei singoli topic e payload o del sorgente ;)

**Comment by gforment** [comment]
ok, grazie. ho recuperato i messaggi in tail sul log di Mosquitto (anche messaggio di discovery) ma speravo in qualcosa di già fatto da qualcun altro.
